extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/devconsole/EventsMonitor.java

```diff
@@ -0,0 +1,129 @@
+package io.quarkus.arc.runtime.devconsole;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.ListIterator;
+
+import javax.enterprise.context.BeforeDestroyed;
+import javax.enterprise.context.Destroyed;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.spi.EventMetadata;
+import javax.inject.Singleton;
+
+// This bean is only registered in the dev mode
```

Comment:
Is it really true? What if you disable the bean removal feature?

Wouldn't it make more sense to remove the annotation and declare the bean with a build item as we do everywhere else?

Reply:
It is true because of https://github.com/quarkusio/quarkus/pull/15040/files#diff-164fa9cb0c9912604128d33ae6c38003a72dde1fef0e77ac5d76af0ee7ee4087R46

The bean removal feature has no effect here.

Reply:
Ah sorry, I misread it as an `UnremovableBeanBuildItem`. High time I start my week-end!